lib/utility/runJoin.js

```diff
@@ -93,9 +93,9 @@ exports.runJoin = function runJoin(Model, isCollection, instances, models, isInn
 					for (var i = 0; i < result.length; i++) {
 						var item = result[i].toJSON();
 						// if its join as "field" then extract just the field value, else the complete set
-						item = model.mergeType === 'field' ? item[model.fieldName] : [item];
```

Comment:
Wont this change the way composite works now? Application's in production may suffer from this change.